src/account/address.cpp

```diff
@@ -7,39 +7,36 @@
 #include <stdexcept>
 
 Address::Address(bytes const &publicKey) :
-        m_pk(publicKey),
-        m_bech32Address()
+        m_pk(publicKey)
 {
     if (publicKey.size() != PUBLIC_KEY_LENGTH)
         throw std::length_error(ERROR_MSG_KEY_BYTES_SIZE);
+
+    m_bech32Address = computeBech32Address();
 }
 
 Address::Address(std::string const &bech32Address) :
-        m_pk(),
         m_bech32Address(bech32Address)
 {
     if (util::bech32::decode(bech32Address).first != hrp)
         throw std::invalid_argument(ERROR_MSG_BECH32);
+
+    m_pk = computePkFromBech32();
 }
 
-Address& Address::operator = (const Address &address)
+bool Address::operator==(const Address &address) const
```

Comment:
In a future PR, please add tests for this as well.

Reply:
Yes, I've just added tests for it in [this](https://github.com/ElrondNetwork/elrond-sdk-erdcpp/pull/7) PR